packages/client/src/base/selection-service.ts

```diff
@@ -53,14 +60,26 @@ export class SelectionService implements SModelRootListener {
     @inject(TYPES.ILogger)
     protected logger: ILogger;
 
-    constructor(@multiInject(TYPES.SelectionListener) @optional() protected selectionListeners: SelectionListener[] = []) {}
+    @multiInject(TYPES.ISelectionListener)
+    @optional()
+    protected selectionListeners: ISelectionListener[] = [];
+
+    @postConstruct()
+    protected initialize(): void {
+        this.selectionListeners.forEach(listener =>
+            this.onSelectionChanged(change => listener.selectionChanged(change.root, change.selectedElements, change.deselectedElements))
+        );
+    }
 
-    register(selectionListener: SelectionListener): void {
-        distinctAdd(this.selectionListeners, selectionListener);
+    @preDestroy()
+    dispose(): void {
+        this.onSelectionChangedEmitter.dispose();
```

Comment:
I'm a big fan of the "something implements Disposable so we have a toDispose collection"-pattern. But that is just a preference so we can keep it as is but I always appreciated that in the framework that I can add the toDipose stuff already when I override initialize somehow.